packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts

```diff
@@ -0,0 +1,130 @@
+import * as fs from 'fs-extra';
+
+import {
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getNpmPath,
+  getNpxPath,
+  initJSProjectWithProfile,
+  nspawn,
+} from '@aws-amplify/amplify-e2e-core';
+import {} from '@aws-amplify/amplify-e2e-core';
+import execa from 'execa';
+import { spawnSync } from 'child_process';
+
+const nextIndexContent = `export default function Home() {
+    return (
+      <div>
+        {Object.entries(process.env.AWS_EXPORTS).map(
+          ([key, value]) => \`\${key}: \${value}\`
+        )}
+      </div>
+    );
+  }  
+`;
+
+const nextConfigContent = `module.exports = () => {
+    const env = {
+      AWS_EXPORTS: (() => require("./src/aws-exports"))(),
+    };
+  
+    return { env };
+  };  
+`;
+
+const es6reactApp = `import { Amplify } from "aws-amplify";
+import awsExports from "./aws-exports";
+
+Amplify.configure(awsExports);
+
+function App() {
+  return (
+    <>
+      {Object.entries(process.env.AWS_EXPORTS).map(
+        ([key, value]) => \`\${key}: \${value}\`
+      )}
+    </>
+  );
+}
+
+export default App;
+`;
+
+describe('test require on aws-exports file on different JS projects', () => {
+  let projRoot: string;
+
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('aws-exports-test');
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
+  });
+
+  test('works with commonJS', async () => {
+    // init next app
+    const appRoot = await createNextApp(projRoot, 'nextjs');
+
+    // install amplify dependency
+    spawnSync(getNpmPath(), ['install', '-E', '@aws-amplify/ui-react', 'aws-amplify'], { cwd: appRoot });
+
+    // amplify init, this will generate aws-exports in the project
+    await initJSProjectWithProfile(appRoot, { name: 'nextjs' });
+
+    // check if amplify init generated project files
+    expect(fs.existsSync(`${appRoot}/amplify/.config/project-config.json`)).toBeTruthy();
+    expect(fs.existsSync(`${appRoot}/src/aws-exports.js`)).toBeTruthy();
+    expect(fs.existsSync(`${appRoot}/next.config.js`)).toBeTruthy();
+
+    fs.writeFileSync(`${appRoot}/pages/index.js`, nextIndexContent, { encoding: 'utf8', flag: 'w' });
+    fs.writeFileSync(`${appRoot}/next.config.js`, nextConfigContent, { encoding: 'utf8', flag: 'w' });
+
+    // this will attempt to compile and start next server.
```

Comment:
if this is spawning next server process then test worker may hang waiting for child process to finish.
Could you check if this is the case? if yes we should either kill the child process or find other command.
If this is working as expected then it's fine.

Reply:
This was working as intended when testing in my machine